### ISSUE TITLE
Not starting

### ISSUE THREAD

**Issue by E34rrsff** [flathub/com.valvesoftware.Steam#1003]
I tried reinstalling Steam 'cause it wouldn't launch after I changed some sandbox permissions with flatpak. I keep getting this when I try running it:

```
INFO:root:https://github.com/flathub/com.valvesoftware.Steam/wiki
ERROR:root:~/.steam/root isn't a symlink to an existing directory, cannot determine current prefix
INFO:root:Will set XDG dirs prefix to /home/$USER/.var/app/com.valvesoftware.Steam
Traceback (most recent call last):
  File "/app/bin/steam-wrapper", line 8, in <module>
    sys.exit(main())
  File "/app/lib/python3.10/site-packages/steam_wrapper.py", line 493, in main
    should_restart += migrate_data(current_info, xdg_dirs_prefix)
  File "/app/lib/python3.10/site-packages/steam_wrapper.py", line 340, in migrate_data
    ignore = _get_host_xdg_mounts("xdg-data", flatpak_info)
  File "/app/lib/python3.10/site-packages/steam_wrapper.py", line 311, in _get_host_xdg_mounts
    dirs.add(os.path.join(*path_seq[1:]))
TypeError: join() missing 1 required positional argument: 'a'
```

**Comment by gasinvein** [comment]
> I changed some sandbox permissions with flatpak

What permissions? You can post `~/.local/share/flatpak/overrides/com.valvesoftware.Steam` file contents here.

**Comment by E34rrsff** [comment]
It was just some permissions to allow flatpak apps access my GTK themes.

Output:

```
[Context]
unset-environment=GTK_THEME;
filesystems=!/usr/share/fonts;!/home/$USER/.icons;!/usr/share/themes;!/usr/share/icons;!/home/$USER/.fonts;!/home/$USER/.themes;

[Environment]
GTK_THEME=
```

**Comment by gasinvein** [comment]
And what's in `~/.local/share/flatpak/overrides/global`?

**Comment by E34rrsff** [comment]
Output:

```
xdg-music:ronxdg-pictures:ronxdg-run/app/com.discordapp.Discord:createn!/home/$USER/.themesn!/usr/share/fontsn!/home/$USER/.iconsn!/home/$USER/.fontsn!/usr/share/iconsn!/usr/share/themesn
```
Commenting out those settings does make it work again, but how can I keep them on and have Steam working?

**Comment by gasinvein** [comment]
> Output:
> 
> ```
> xdg-music:ronxdg-pictures:ronxdg-run/app/com.discordapp.Discord:createn!/home/$USER/.themesn!/usr/share/fontsn!/home/$USER/.iconsn!/home/$USER/.fontsn!/usr/share/iconsn!/usr/share/themesn
> ```

Sorry, I've meessed out the command, probably escaping like this didn't work in `bash`. I've fixed it and updated my previous post, can you please run it again?

> Commenting out the settings in the 'global' file does make it work again, but how can I keep them on and have Steam working?

You have a lot of override, some of them are useless, and some must be causing this issue. We should figure out which one.

**Comment by E34rrsff** [comment]
Output:

```
xdg-music:ro
xdg-pictures:ro
xdg-run/app/com.discordapp.Discord:create
!/home/$USER/.themes
!/usr/share/fonts
!/home/$USER/.icons
!/home/$USER/.fonts
!/usr/share/icons
!/usr/share/themes
```

I thought `flatpak override --reset com.valvesoftware.Steam` would fix the issue. This only happened after adding permissions for access to my icons, themes, and fonts folders; unsure if those other permissions on top are on by default. I would like to keep them on, though.

**Comment by gasinvein** [comment]
Something is off here. All the custom permissions start with `!`, meaning they're _disallowing_  access to those paths. The traceback suggests it happened when processing a `xdg-data` permission, but there are none in the provided overrides. And where does the `$USER` come from? Is it a literal string, or you've censored your username this way?

**Comment by E34rrsff** [comment]
The `!`s may have been from me messing with `~/.local/share/flatpak/overrides/global` or something else, but I don't remember.... 

I'm unsure what exactly caused the issue, and although I thought I had done this before, ensuring all permissions are gone with `flatpak override --reset`, `flatpak --user override --reset`, & `sudo flatpak override --reset` fixes it for me.

I then added my permissions again, while only using `--user` and using `~` instead of the full path to my home directory. Steam still works, and running the command from your post again gives me this:

```
~/.themes
xdg-music:ro
xdg-pictures:ro
xdg-run/app/com.discordapp.Discord:create
~/.fonts
~/.icons
/usr/share/fonts
/usr/share/themes
/usr/share/icons
```

 Thanks for the help. It was most likely some silly mistake I did, and I did replace the username with '`$USER`' in my comments.
